main.go

```diff
@@ -3,11 +3,20 @@ package main
 import (
 	"os"
 
+	"flag"
+	"fmt"
 	"github.com/Sirupsen/logrus"
 	"github.com/rancher/agent/events"
 )
 
+var Version = "v0.1.5-agent"
+
 func main() {
+	if len(os.Args) > 1 {
```

Comment:
You shouldn't check the length.  Just make the flag a global and make it a bool flag.  If some body runs `agent -version` then the value of the bool flag will be true and then you print the version and exit.